src/cmap/connect.ts

```diff
@@ -435,78 +414,71 @@ function loadSocks() {
   return socks;
 }
 
-function makeSocks5Connection(options: MakeConnectionOptions, callback: Callback<Stream>) {
+async function makeSocks5Connection(options: MakeConnectionOptions): Promise<Stream> {
   const hostAddress = HostAddress.fromHostPort(
     options.proxyHost ?? '', // proxyHost is guaranteed to set here
     options.proxyPort ?? 1080
   );
 
   // First, connect to the proxy server itself:
-  makeConnection(
-    {
-      ...options,
-      hostAddress,
-      tls: false,
-      proxyHost: undefined
-    },
-    (err, rawSocket) => {
-      if (err || !rawSocket) {
-        return callback(err);
-      }
+  const rawSocket = await makeSocket({
+    ...options,
+    hostAddress,
+    tls: false,
+    proxyHost: undefined
+  });
 
-      const destination = parseConnectOptions(options) as net.TcpNetConnectOpts;
-      if (typeof destination.host !== 'string' || typeof destination.port !== 'number') {
-        return callback(
-          new MongoInvalidArgumentError('Can only make Socks5 connections to TCP hosts')
-        );
-      }
+  const destination = parseConnectOptions(options) as net.TcpNetConnectOpts;
+  if (typeof destination.host !== 'string' || typeof destination.port !== 'number') {
+    throw new MongoInvalidArgumentError('Can only make Socks5 connections to TCP hosts');
+  }
 
-      try {
-        socks ??= loadSocks();
-      } catch (error) {
-        return callback(error);
+  socks ??= loadSocks();
+
+  try {
+    // Then, establish the Socks5 proxy connection:
+    const { socket } = await socks.SocksClient.createConnection({
+      existing_socket: rawSocket,
+      timeout: options.connectTimeoutMS,
+      command: 'connect',
+      destination: {
+        host: destination.host,
+        port: destination.port
+      },
+      proxy: {
+        // host and port are ignored because we pass existing_socket
+        host: 'iLoveJavaScript',
+        port: 0,
+        type: 5,
+        userId: options.proxyUsername || undefined,
```

Comment:
Is setting `undefined` explicitly necessary? If `proxyUsername` or `proxyPassword` are not in the options, their value will be `undefined` anyway.

Reply:
If proxyUsername is an empty string we pass in `undefined` instead